addon/index.js

```diff
@@ -0,0 +1,14 @@
+import useMachine from './usables/use-machine';
+import { matchesState as xstateMatchesState } from 'xstate';
+
+function matchesState(state, statechartPropertyName = 'statechart') {
+  return function () {
+    return {
+      get() {
+        return xstateMatchesState(state, this[statechartPropertyName].currentState.value);
```

Comment:
this is still the only thing that doesn't sit right with me -- mostly because it's relying on an implicit property name.

I wonder if it makes sense to do something like:
```ts
class Foo {
  @use myInterpreter = useMachine(machineStuff);

  @matches(this.myInterpreter, 'someState');
  isSomeState;
}
```
idk.

Reply:
This is how this works already.

```js
class Foo {
  @use myInterpreter = useMachine(myMachine);

  @matchesState('busy', 'customStatechartName')
  isBusy;
}
```

If you leave out the default property it checks the `statechart`-property. I think best ist to make this property name configurable via a setting.

Reply:
I think if possible to pass a 'this' property to a decorator, that would make the most sense intuitively, but I don't know if the decorators are evaluated before instantiation.

My gut on this is that matchesState is for a very imperitive style of consuming statecharts, and I think we should survey some people outside of the xstate community for feedback :/

Reply:
I don't follow what you mean by imperative style for consuming. `matchesState` is reactive getter for matching against state patterns - it's just a decorator that makes it more ergonomical (arguably) to match against the current-state of the interpreter.

You could also do this when you want to react to state changes but `@matchesState` saves you some typing.

```
get isBusy() {
  return matchesState('busy', this.statechart.state);
}
```

Reply:
> I don't follow what you mean by imperative style for consuming.

Just that there is a magic, undeclared relationship with the property name of the useMachine value.

> You could also do this when you wanted to react to state changes but @matchesState saves you some typing.

I think that's more inline with modern JS. Though, I'd switch the args:
```ts
get isBusy() {
  return matchesState(this.statechart.state.value, 'busy');
}
```
or even:
```ts
get isBusy() {
  return this.statechart.state.match('busy');
}
```